### PR TITLE
[ART-3152] Revisit plashet for-assembly mechanics

### PR DESCRIPTION
**Pull request by vfreex**
Removes the following steps:
- creating a repo based off of tags
- pulling RPMs from all images in an assembly

Adds a new argument `--el-version` because plashet needs to know the
RHEL version to populate a repo.

### CODE REVIEW

File: doozerlib/cli/config_plashet.py [hunk 1]
```diff
@@ -738,7 +738,14 @@ def for_assembly(config: SimpleNamespace, image: Optional[str], rhcos: bool, bre
     runtime.initialize(mode="both", clone_source=False, clone_distgits=False, prevent_cloning=True)
     if not runtime.assembly_basis_event:
         raise DoozerFatalError("for-assembly must be run for an assembly with a basis event")
-    logger.warning(f'Constraining rpm search to stream assembly due to assembly basis event {runtime.assembly_basis_event}')
+    if not el_version:
+        if image:
+            el_version = runtime.image_map[image].branch_el_target()
+        elif rhcos:
+            el_version = 8  # FIXME: Currently RHCOS is RHEL8 based. Revisit here when this changes
```
Comment (jupierce) [inline doozerlib/cli/config_plashet.py:745]: Interesting.. seems like elX should not be part of the rhcos dependencies schema. I'll make a ticket to refactor this. After that, should can blindly include whatever is in the list.

Comment (jupierce) [reply doozerlib/cli/config_plashet.py:745]: https://issues.redhat.com/browse/ART-3156

Comment (vfreex) [reply doozerlib/cli/config_plashet.py:745]: Currently all dependencies (including group, image member, and rhcos) are following this format:
```yaml
dependencies:
   rpms:
   - elN: NVR
   - elN: NVR
```

Looks like only group dependencies need el version. What do you think if we change the schema for image member and rhcos dependencies to the way we specify "is"?
```
dependencies:
    rpms:
    - nvr: NVR

```

Comment (vfreex) [reply doozerlib/cli/config_plashet.py:745]: However, it doesn't seem to me that ART-3156 will eventually eliminate the hardcoded version. It is needed to populate group dependencies. Another place I hardcode the RHCOS RHEL version is https://github.com/openshift/aos-cd-jobs/pull/2690/files#diff-166229d927199bef1ae96efc6b610a1924440315ff92ba7de0527fc2c83f3498R107, for the same reason.

Comment (vfreex) [reply doozerlib/cli/config_plashet.py:745]: Or, maybe we can define the RHEL version of RHCOS in ocp-build-data?

**Review by jupierce — APPROVED** [review]
Nits. We can hold off on eliminating hardcoding el version for rhcos/images.

File: doozerlib/cli/config_plashet.py [hunk 2]
```diff
@@ -715,21 +716,20 @@ def is_embargoed(an_nvre):
...
 @click.option('--poll-for', default=15, type=click.INT, help='Allow up to this number of minutes for auto-signing')
-def for_assembly(config: SimpleNamespace, image: Optional[str], rhcos: bool, brew_tags: Tuple[Tuple[str, str], ...], inherit: bool, signing_advisory_id: Optional[int], signing_advisory_mode: str, poll_for: int):
+def for_assembly(config: SimpleNamespace, image: Optional[str], rhcos: bool, el_version: Optional[int], signing_advisory_id: Optional[int], signing_advisory_mode: str, poll_for: int):
     """
-    Creates a directory containing arch specific yum repository subdirectories based on a complete set of RPMs required for image builds.
-    In other words, when doozer runs brew builds for an assembly image, it should only need to pass in a single repository created by for-assembly in order for the image to build successfully.
+    Creates a directory containing arch specific yum repository subdirectories based on RPMs explicitly listed as dependencies in the assembly
+    (whether for the whole assemble, rhcos, or a specific image, depending on the command line invocation).
```
[the remark this answers is not in the document]
Comment (vfreex) [reply doozerlib/cli/config_plashet.py:731]: Thanks.
